### ISSUE TITLE
Build breakage ‘DNS_ERR_CANCEL’ undeclared

### ISSUE THREAD

**Issue by denik** [gevent/gevent#60]
<b>What version of the gevent are you using?</b>
latest (hg)

<b>What version of libevent are you using?</b>
1.4.2

<b>On what operating system?</b>
Ubuntu 9.10

<b>On what Python?</b>
2.6

python setup.py build
running build
running build_py
running build_ext
building 'gevent.core' extension
gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.6 -c gevent/core.c -o build/temp.linux-x86_64-2.6/gevent/core.o
gevent/core.c:671: warning: ‘struct evutil_addrinfo’ declared inside parameter list
gevent/core.c:671: warning: its scope is only this definition or declaration, which is probably not what you want
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_new’:
gevent/core.c:3050: warning: implicit declaration of function ‘evdns_base_new’
gevent/core.c:3050: warning: assignment makes pointer from integer without a cast
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_free’:
gevent/core.c:3288: warning: implicit declaration of function ‘evdns_base_free’
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_add_nameserver’:
gevent/core.c:3343: warning: implicit declaration of function ‘evdns_base_nameserver_ip_add’
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_count_nameservers’:
gevent/core.c:3459: warning: implicit declaration of function ‘evdns_base_count_nameservers’
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_set_option’:
gevent/core.c:3698: warning: implicit declaration of function ‘evdns_base_set_option’
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_getaddrinfo’:
gevent/core.c:3813: error: storage size of ‘__pyx_v_hints’ isn’t known
gevent/core.c:4195: warning: implicit declaration of function ‘evdns_getaddrinfo’
gevent/core.c:4195: warning: assignment makes pointer from integer without a cast
gevent/core.c:3813: warning: unused variable ‘__pyx_v_hints’
gevent/core.c: In function ‘__pyx_pf_6gevent_4core_10evdns_base_resolve_ipv4’:
gevent/core.c:4373: warning: implicit declaration of function ‘evdns_base_resolve_ipv4’
gevent/core.c:4373: warning: assignment makes pointer from integer without a cast
gevent/core.c: At top level:
gevent/core.c:4483: warning: ‘struct evutil_addrinfo’ declared inside parameter list
gevent/core.c:4483: error: conflicting types for ‘__pyx_f_6gevent_4core___getaddrinfo_handler’
gevent/core.c:671: note: previous declaration of ‘__pyx_f_6gevent_4core___getaddrinfo_handler’ was here
gevent/core.c: In function ‘__pyx_f_6gevent_4core___getaddrinfo_handler’:
gevent/core.c:4625: error: ‘EVUTIL_EAI_CANCEL’ undeclared (first use in this function)
gevent/core.c:4625: error: (Each undeclared identifier is reported only once
gevent/core.c:4625: error: for each function it appears in.)
gevent/core.c:4775: error: dereferencing pointer to incomplete type
gevent/core.c:4785: error: dereferencing pointer to incomplete type
gevent/core.c:4808: error: dereferencing pointer to incomplete type
gevent/core.c:4818: error: dereferencing pointer to incomplete type
gevent/core.c:4828: error: dereferencing pointer to incomplete type
gevent/core.c:4848: error: dereferencing pointer to incomplete type
gevent/core.c:4848: error: dereferencing pointer to incomplete type
gevent/core.c:4848: error: dereferencing pointer to incomplete type
gevent/core.c:4877: error: dereferencing pointer to incomplete type
gevent/core.c: In function ‘__pyx_f_6gevent_4core___evdns_handler’:
gevent/core.c:5275: error: ‘DNS_ERR_CANCEL’ undeclared (first use in this function)
error: command 'gcc' failed with exit status 1

<p><a href="http://code.google.com/p/gevent/issues/detail?id=60">Reported by pschork.</a></p><hr/><h2>earlier comments</h2>
<p><strong>pschork said, at 2010-12-21T00:46:28.000Z:</strong></p>

<p>fwiw, I can build 0.13.1 just fine</p><p><strong>Denis.Bilenko said, at 2010-12-21T05:07:00.000Z:</strong></p>

<p>Currently trunk requires libevent-2 to run.</p><p><strong>Denis.Bilenko said, at 2011-12-20T18:07:00.000Z:</strong></p>

<p>This happend on trunk when trunk was on libevent2. It's no longer the case, so this is no longer relevant.</p>
